File: BIAN_YUAN_Code/485.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        Still using little DP trick, i.e. to memorize the maximum 1s end with this component       
        """
        leng=len(nums)
        d=[None]*leng
        
        d[0]=nums[0]
        
        if leng==1:
            return d[0]
        
        for i in range(1, leng):
            if nums[i]==0:
                d[i]=0
            else:
                d[i]=1+d[i-1]
                
        return max(d)

File: BIAN_YUAN_Code/485.java
/**
 * 计算int数组 最长连续1的长度
 * 
 * @author tianning.htn
 *
 */
public class Maths {
	
	/**
	 * 直接遍历，别想什么数组转字符串啊，切割啊，妈的切完还要遍历算长度的啊
	 * 
	 * @param s
	 * @return
	 */
	  public int findMaxConsecutiveOnes(int[] nums) {
	         int max = 0;
			 int current = 0;
			 for (int i : nums) {
				if (i == 1) {
					current++;
				}else if (i == 0) {
					max = Math.max(max, current);
					current = 0;
				}
			}
			max = Math.max(max, current);
			return max;
	    }

	
	public static void main(String[] args) {
		System.out.println(findMaxConsecutiveOnes(new int[{1,0,1,0,0,0,1,1,1,1,1,0,1,1,0}]));
	}
}
